enumerables.rb

```diff
@@ -0,0 +1,130 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?(arg = nil)
```

Comment:
## my_all?

you need to work more on my_all? method it's not returning the expected value for numbers, words and falsy values

```ruby

  array = [12, 3, 432, 52, 5, 2, 35, 2, 3, 23, 42, 52, 65, 68, 56]
  
  array.my_all?(3) == array.all?(3) => false

```